### PR TITLE
Check attached for snapshot last and first

### PR DESCRIPTION
**Pull request by beeender**
Results's last() and first() should return none if the row is deleted.

### CODE REVIEW

File: src/results.cpp [hunk 1]
```diff
@@ -215,7 +215,9 @@ util::Optional<RowExpr> Results::first()
         case Mode::Query:
         case Mode::TableView:
             update_tableview();
-            return m_table_view.size() == 0 ? util::none : util::make_optional(m_table_view.front());
+            return (m_table_view.size() == 0) ||
+                (m_update_policy == UpdatePolicy::Never && !m_table_view.is_row_attached(0)) ?
+                util::none : util::make_optional(m_table_view.front());
```
Comment (beeender) [inline src/results.cpp]: maybe it makes more sense to return a `util::make_optional(RowExpr())` instead? so it returns `none` for empty snapshot, but empty Row for deleted rows?

Comment (bdash) [inline src/results.cpp]: Returning a detached row accessor would be consistent with the behavior of `Results::get`.

File: tests/results.cpp [hunk 2]
```diff
@@ -2038,6 +2038,22 @@ TEST_CASE("results: snapshots") {
...
+            table->add_empty_row();
+        });
+        Results results(r, *table);
+        auto snapshot = results.snapshot();
+        write([=] {;
+            table->clear();
+        });
+
+        REQUIRE_FALSE(snapshot.get(0).is_attached());
+        REQUIRE_FALSE(snapshot.first().value().is_attached());
```
Comment (bdash) [inline tests/results.cpp]: Typically we'd write `->is_attached()` rather than `.value().is_attached()`.

File: src/results.cpp [hunk 3]
```diff
@@ -215,7 +215,13 @@ util::Optional<RowExpr> Results::first()
         case Mode::Query:
         case Mode::TableView:
             update_tableview();
-            return m_table_view.size() == 0 ? util::none : util::make_optional(m_table_view.front());
+            if (m_table_view.size() == 0) {
+                return util::none;
+            } else if (m_update_policy == UpdatePolicy::Never && !m_table_view.is_row_attached(0)) {
+                return util::make_optional(RowExpr());
```
Comment (bdash) [inline src/results.cpp]: Calling `util::make_optional` isn't necessary here. It was only necessary in the ternary expression as it allows each branch needs to have the same type.

File: src/results.cpp [hunk 4]
```diff
@@ -215,7 +215,13 @@ util::Optional<RowExpr> Results::first()
         case Mode::Query:
         case Mode::TableView:
             update_tableview();
-            return m_table_view.size() == 0 ? util::none : util::make_optional(m_table_view.front());
+            if (m_table_view.size() == 0) {
+                return util::none;
+            } else if (m_update_policy == UpdatePolicy::Never && !m_table_view.is_row_attached(0)) {
```
Comment (bdash) [inline src/results.cpp]: Minor nit, but in object store we typically don't use braces on single-line if statements. It's also not really necessary to have an `else` when the `if`'s body ended with a `return`.

**Pull request merged**